content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b2 register.
+    };
+```
+### Example 3: (Error case)
+This error case is very similar to Example 2, but `m_texSRV_unbounded` is declared after `m_texSRV1`, but before `m_texSRV2`.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t0 register.
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t1 register, and will take over t2+.
+        Texture2D<float4>        m_texSRV2;      // ERROR. There's no tN available to bind this resource to.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u0 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u1 register.
+        Sampler                  m_sampler1;     // Will be bound to s0 register.
+        Sampler                  m_sampler2;     // Will be bound to s1 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b2 register.
+    };
+```
+### Example 4:
+This case fixes the issues in Example 3. By moving the point of declaration of `m_texSRV_unbounded[]` after `m_texSRV2` there are no more register assignment conflicts.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t0 register.
+        Texture2D<float4>        m_texSRV2;      // Will be bound to t1 register.
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t2 register, and will take over t3+.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u0 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u1 register.
+        Sampler                  m_sampler1;     // Will be bound to s0 register.
+        Sampler                  m_sampler2;     // Will be bound to s1 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b2 register.
+    };
```

Comment:
```suggestion
ShaderResourceGroupSemantic slot1
{
    FrequencyId = 1;
};

struct MyStruct
{
    float3 m_a;
    float3 m_b;
};

//For DX12, --unique-idx is OFF, so the expected register assignment is as follows
ShaderResourceGroup SRG1 : slot1
{
    Texture2D<float4>        m_texSRV1;             // Binds to t0 register.
    Texture2D<float4>        m_texSRV2;             // Binds to t1 register.
    Texture2D<float4>        m_texSRV_unbounded[];  // Binds to t2 register, and takes over t3+.
    RWTexture2D<float4>      m_texUAV1;             // Binds to u0 register.
    RWTexture2D<float4>      m_texUAV2;             // Binds to u1 register.
    Sampler                  m_sampler1;            // Binds to s0 register.
    Sampler                  m_sampler2;            // Binds to s1 register.
    ConstantBuffer<MyStruct> m_cb1;                 // Binds to b1 register. (Because the ConstantBuffer reserved for SRG1 binds to b0 register).
    ConstantBuffer<MyStruct> m_cb2;                 // Binds to b2 register.
};
```
* Remove indent. It's not necessary when defining within a code snippet (```cpp). This also allows longer comments to fit on the page.
* Simplified "will be bound to" to "binds to".

Reply:
Fixed.